L1Trigger/TrackFindingTracklet/src/MatchProcessor.cc

```diff
@@ -671,6 +694,19 @@ bool MatchProcessor::matchCalculator(Tracklet* tracklet, const Stub* fpgastub, b
                                    << " " << drcut / settings_.krprojshiftdisk() << " r = " << stub->r();
     }
 
+    bool keep = true;
+    if (!settings_.doKF() || !settings_.doMultipleMatches()) {
+      // Case of allowing only one stub per track per layer (or no KF which implies the same).
+      if (imatch && tracklet->match(layerdisk_)) {
+        // Veto match if is not the best one for this tracklet (in given layer)
+        auto res = tracklet->resid(layerdisk_);
+        keep = abs(ideltaphi) < abs(res.fpgaphiresid().value());
+        imatch = keep;
+      }
+    }
+    if (not keep)
+      match = false;  // FIX: should calc keep with float point here.
```

Comment:
match & imatch both indicates it a stub matches a track candidate and should be kept. But imatch is based on a full digitital emulation of the firmware, and match is just a floating point calculation. If the decision is taken at L703 not to keep the stub (keep = false), then imatch is set false at L704 and match is set false at L708. So both match & imatch become false. This is correct (although it neglects the possibility that the floating point calculation might lead to slightly different stubs being kept.
@bryates or @aryd can you confirm this please?

Reply:
Yes, this is correct. I've never looked into edge cases where the floating point calculation might slightly disagree and give a match. @aryd did you investigate this when first created this logic?

Reply:
Sorry, but after this statement `match` is never checked any more (correct me if I am wrong), while `imatch` is.
The problem I see here is that you are updating an unused variable, which is not what you were supposing to do, I guess

Reply:
Ian, I had misunderstood your comment when I responded yesterday… 

 Regarding the comment about:

  if (not keep) match = false;  // FIX: should calc keep with float point here.

I don’t think I created this logic, if you turn on the ‘blame’ in git you see that this code was introduced with the commit “Fix stub order bug”. 

The variable match is calculated earlier and is really only used in a debug printout - but it is helpful when you want to check the fixed size integer calculation vs. the full floating point. 

But this extra if statement above has no function since matches never used later in the code.

I suggest we remove it.

Reply:
I have removed this code now and pushed the changes.